### PR TITLE
csharp: add test messages for test results

### PR DESCRIPTION
**Pull request by ErikSchierboom**
@iHiD Let me know what you think.

### CODE REVIEW

File: exercises/csharp-2-a/.meta/test_messages.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    "cmd": "Sentence.WordWithMostVowels(\"tea\")"
+  },
+  "LongestWordIsNotWordWithMostVowels": {
+    "cmd": "Sentence.WordWithMostVowels(\"the area of a circle\")"
+  },
+  "LastWordIsWordWithMostVowels": {
+    "cmd": "Sentence.WordWithMostVowels(\"extremely fresh vegetables\")"
+  },
+  "MixedCasing": {
+    "cmd": "Sentence.WordWithMostVowels(\"TEST YOUR MIGHT\")"
```
Comment (ErikSchierboom) [inline exercises/csharp-2-a/.meta/test_messages.json]: @iHiD Can you spot the pop-culture reference here?

Comment (iHiD) [reply exercises/csharp-2-a/.meta/test_messages.json]: Nope! :)

Comment (iHiD) [reply exercises/csharp-2-a/.meta/test_messages.json]: Apparently mortal compact. I never had a console with it on growing up!

Comment (ErikSchierboom) [reply exercises/csharp-2-a/.meta/test_messages.json]: You didn't? Oh wow. It is indeed Mortal Kombat. I played that game so much 👊

Comment (iHiD) [reply exercises/csharp-2-a/.meta/test_messages.json]: I wish I had.

File: README.md [hunk 2]
```diff
@@ -18,6 +18,46 @@ These are examples of valid exercise names:
 - `javascript-2-b` : second part of exercise B for JavaScript.
 - `ruby-1-b` : first part of exercise B for Ruby.
 
+### Test messages
+
+Solving exercises that are part of the experiment is different from the regular website's process in two ways:
+
+1. Exercises are solved exclusively using in-browser coding.
+1. The test suite used to test the code is not available to the user.
+
```
Comment (iHiD) [inline README.md]: How about adding something here saying something like:

----

This means that when the code does _not_ pass one of the tests, the user won't see the code of the test that failed, or know with what input the code was called. To fix this, we need to provide a bit of context when we show errors to the user. 

For an assert_equal failure, the error with context might look as follows:

> We tried running `MyTests.MyTestMethod(1)` but got the following error:
> ```
> Expected: "foobar"
>   Actual: "barfood"
> ```

File: README.md [hunk 3]
```diff
@@ -18,6 +18,46 @@ These are examples of valid exercise names:
...
+This means that when the code does _not_ pass one of the tests, the user won't know with what input the code was called. To help with this, each exercise should define a `.meta/test_messages.json` file.
```
Comment (iHiD) [inline README.md]: Add something like above then change this to?

```suggestion
To achieve with this, each exercise should define a `.meta/test_messages.json` file.
```

and then merge that with the paragraph below?

File: README.md [hunk 4]
```diff
@@ -18,6 +18,46 @@ These are examples of valid exercise names:
...
+
+```json
+{
+  "MyTestMethod": {
+    "msg": "We ran `MyTests.MyTestMethod(1)` and got the following error:\n\n%{output}"
+  }
+}
+```
+
+Note that you can embed the [test runner's][test-runner-output-format] `"message"` value (usually the assertion error message) in the message by using `%{output}`.
```
Comment (iHiD) [inline README.md]:
```suggestion
Note that you can embed the [test runner's][test-runner-output-format] output (usually the assertion error message) in the message by using `%{output}`.
```

File: README.md [hunk 5]
```diff
@@ -18,6 +18,55 @@ These are examples of valid exercise names:
...
 - `ruby-1-b` : first part of exercise B for Ruby.
 
+### Test messages
+
+Solving exercises that are part of the experiment is different from the regular website's process in two ways:
+
+1. Exercises are solved exclusively using in-browser coding.
+1. The test suite used to test the code is not available to the user.
+
+This means that when the code does _not_ pass one of the tests, the user won't see the code of the test that failed, or know with what input the code was called. To fix this, we need to provide a bit of context when we show errors to the user.
```
Comment (iHiD) [inline README.md]:
```suggestion
This means that when the code does _not_ pass one of the tests, the user won't see the code of the test that failed or the input the code was called. To fix this, we need to provide a bit of context when we show errors to the user.
```

**Pull request merged**